src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,90 @@
 import { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+type Props = {
+  onAdd: (movie: Movie) => void;
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const [count, setCount] = useState(0);
+  const [descriptionField, setDescriptionField] = useState('');
+  const [titleField, setTitleField] = useState('');
+  const [imgUrlField, setImgUrlField] = useState('');
+  const [imdbUrlField, setImdbUrlField] = useState('');
+  const [imdbIdField, setImdbIdField] = useState('');
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    const movieToUpload = {
+      title: titleField,
+      description: descriptionField,
+      imgUrl: imgUrlField,
+      imdbUrl: imdbUrlField,
+      imdbId: imdbIdField,
+    };
```

Comment:
instead of writing too much code in your `movieToUpload` object, you could change names in your `useState` variables so it matches names in both cases
```suggestion
  const [count, setCount] = useState(0);
  const [description, setDescription] = useState('');
  const [title, setTitle] = useState('');
  const [imgUrl, setImgUrl] = useState('');
  const [imdbUrl, setImdbUrl] = useState('');
  const [imdbId, setImdbId] = useState('');

  const handleSubmit = (event: React.FormEvent) => {
    event.preventDefault();

    const movieToUpload = {
      title,
      description,
      imgUrl,
      imdbUrl,
      imdbId,
    };
```